### PR TITLE
Fix deprecations

### LINKED ISSUE #2049: Fix DeprecationWarnings
- logging.warn -> logging.warning
- astropy.modeling.fitting.LMLSQFitter -> TRFLSQFitter

### PR DESCRIPTION
**Pull request by nden**
Resolves [HLA-1531](https://jira.stsci.edu/browse/HLA-1531)

Closes #2049

This PR addresses two DeprecationsWarnings in the regression tests. 
- Replaced log.warn with log.warning
- Replaced LMLSQFitter with TRFLSQFitter

Regression tests pass:
https://github.com/spacetelescope/RegressionTests/actions/runs/16352978260/job/4620432714

**Checklist for maintainers**
- [ ] added entry in `CHANGELOG.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant label(s)

### CODE REVIEW

File: drizzlepac/haputils/astrometric_utils.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
                            sigma_clipped_stats, SigmaClip)
 from astropy.visualization import SqrtStretch
 from astropy.visualization.mpl_normalize import ImageNormalize
-from astropy.modeling.fitting import LMLSQFitter
+from astropy.modeling.fitting import TRFLSQFitter
```
Comment (mdlpstsci) [inline drizzlepac/haputils/astrometric_utils.py:46]: I have a branch where I have fixed this.  However, I had Matt run a large statistical test generating SVMs so that I can examine the results with regard to this change.  In the past, changing the **fitter** can impact the quality of the results in unexpected ways.  I also am printing out the resultant "fit" table (only for the test) to see if there is a better variable to base the **_best_** fit on aside from the **flux_fit** maximum.  As such, I would rather this not be a part of this PR (unless you feel strongly).

File: drizzlepac/haputils/astrometric_utils.py [hunk 2]
```diff
@@ -868,8 +868,8 @@ def find_fwhm(psf, default_fwhm, log_level=logutil.logging.INFO):
     # 2.5 x the background value can be considered part of a source.
     iraffind = DAOStarFinder(threshold=2.5 * mmm_bkg(psf), fwhm=default_fwhm)
     # The LevMarLSQFitter fitter is no longer recommended. To use the
-    # Levenberg-Marquardt algorithm without bounds, use LMLSQFitter.
-    fitter = LMLSQFitter()
+    # Levenberg-Marquardt algorithm without bounds, use TRFLSQFitter.
+    fitter = TRFLSQFitter()
```
Comment (mdlpstsci) [inline drizzlepac/haputils/astrometric_utils.py:872]: Same comment as the import statement.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
Request a change in astrometric_utils.py module.  This PR also has changes encapsulated in PR#2047.

File: drizzlepac/haputils/astrometric_utils.py (hunk 1 above)
Comment (nden) [reply drizzlepac/haputils/astrometric_utils.py:46]: That's fine, I can remove it from this PR. It's likely it will be removed in the next astropy release, some time in December.

Comment (nden) [reply drizzlepac/haputils/astrometric_utils.py]: Done

**Comment by nden** [comment]
@mdlpstsci This PR was based on #2047. Now that 2047 is merged and I rebased this one, it should be good for review again.
